charmhelpers/contrib/openstack/context.py

```diff
@@ -2181,9 +2181,64 @@ def __call__(self):
 class HostInfoContext(OSContextGenerator):
     """Context to provide host information."""
 
+    def _get_use_fqdn_hint(self):
+        """Get hint from local unit KV store for whether FQDN should be used
+
+        Depending on the workload a charm manages, the use of FQDN vs.
+        shortname may be a deploy-time decision, i.e. behavour can not change
+        on post deploy charm upgrade or configuration change.
+
+        The hint is passed on as a flag in the context to allow the decision to
+        be made in the Jinja2 configuration template.
+
+        :returns: True or False
+        :rtype: bool
+        """
+        db = kv()
+        return db.get('openstack-hostinfocontext-use-fqdn-hint', False)
```

Comment:
Does this get set in the charm code? I don't see anywhere in charm-helpers where it gets set to True.

Reply:
It is in the charm itself: https://review.opendev.org/#/c/701928/3/hooks/nova_compute_hooks.py

Reply:
@thedac as @rodrigogansobarbieri pointed out it is set by the charm, but your question spurred a thought that the context should really not force the charm authors hand as to how it passes on information for the `use_fqdn_hint`.

So based on your feedback I have changed the context to get this from a callback to the charm code instead.